File: frontend/src/app/dtos/betDto.ts
import { UserDetail } from "./auth-request";
import { ProductDto } from "./productDto";

export class BetDto {
  constructor (
    public id?: number,
    public description?: string,
    public amount?: number,
    public date?:Date,
    public accepted?:boolean,
    public rejected?:boolean,
    public user?: UserDetail,
    public product?: ProductDto
  ) {}
}

File: backend/src/main/java/com/example/backend/Endpoints/dto/BetDto.java
package com.example.backend.Endpoints.dto;

import com.example.backend.Entity.ApplicationUser;
import com.example.backend.Entity.Product;
import io.soabase.recordbuilder.core.RecordBuilder;
import jakarta.validation.constraints.AssertTrue;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import org.springframework.cglib.core.Local;

import java.time.LocalDate;
import java.util.Date;

@RecordBuilder
public record BetDto(
        Long id,
        @Size(max = 100, message = "Description cannot exceed 100 characters")
        String description,
        @NotNull(message = "Amount is required")
        @Min(0)
        Double amount,
        LocalDate date,
        boolean accepted,
        boolean rejected,

        UserDetailDto user,
        @NotNull(message = "Product is required")
        ProductDto product


) {
        @AssertTrue(message = "Accepted and rejected cannot be true at the same time")
        boolean isNotBothAcceptedAndRejected() {
                return !(accepted && rejected);
        }

}
